pkg/utils/kubernetes/kubernetes.go

```diff
@@ -83,3 +85,23 @@ func ObjectMeta(namespaceOrName string, nameOpt ...string) metav1.ObjectMeta {
 	namespace, name := nameAndNamespace(namespaceOrName, nameOpt...)
 	return metav1.ObjectMeta{Namespace: namespace, Name: name}
 }
+
+// WaitUntilResourceDeleted deletes the given resource and then waits until it has been deleted. It respects the
+// given interval and timeout.
+func WaitUntilResourceDeleted(ctx context.Context, client client.Client, obj runtime.Object, namespace, name string, interval, timeout time.Duration) error {
```

Comment:
Why use both a `context` and a `timeout`? It should only be one of both (`context` definitely preferred).